n2y/plugins/footnotes.py

```diff
@@ -0,0 +1,88 @@
+import re
+import logging
+
+from pandoc.types import Note, Str, Para
+
+from n2y.rich_text import TextRichText
+from n2y.blocks import ParagraphBlock
+from n2y.errors import UseNextClass
+
+
+plugin_data_key = "n2y.plugins.footnotes"
+
+logger = logging.getLogger(__name__)
+
+
+class ParagraphWithFootnoteBlock(ParagraphBlock):
+    def __init__(self, client, notion_data, page, get_children=True):
+        super().__init__(client, notion_data, page, get_children)
+        if self._is_footnote():
+            self._attach_footnote_data()
+        else:
+            raise UseNextClass()
+
+    def to_pandoc(self):
+        return None if self._is_footnote() else super().to_pandoc()
+
+    def _attach_footnote_data(self):
+        if plugin_data_key not in self.client.plugin_data:
+            self.client.plugin_data[plugin_data_key] = {}
+        if self._footnote() not in self.client.plugin_data[plugin_data_key]:
+            self.client.plugin_data[plugin_data_key][self._footnote()] = self._footnote_ast()
```

Comment:
Is the result of `self._footnote()` just the number of the foot note? If so, it seems like you won't be able to analyze multple pages that have footnotes since they'll both re-use the same numbers.